<!DOCTYPE html PUBLIC "-//w3c//dtd html 4.0 transitional//en">
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html;
      charset=ISO-8859-1">
    <meta name="Author" content="Tom Hall">
    <meta name="GENERATOR" content="Mozilla/4.8 [en] (Windows NT 5.0; U)
      [Netscape]">
    <title>BioEdit v7.2.6.1 Changes</title>
  </head>
  <body alink="#0000EE" background="back.gif" bgcolor="#FAFFFC"
    link="#0000EE" text="#000000" vlink="#551A8B">
    &nbsp;
    <br>
    <table cellpadding="0" cellspacing="0">
      <caption><br>
      </caption><tbody>
      </tbody>
      <tbody>
        <tr>
          <td align="CENTER"><img src="clear.gif" height="70" width="70"></td>
          <td>
            <center><b><i><font color="#FF0000"><font size="+3">BioEdit
                      v7.2.6.1</font></font></i></b><br>
              <i><font color="#FF0000"><font size="+2">Bug Fixes and
                    Changes</font></font></i>
              <p><b><i><font color="#FF0000"><font size="+0">last update
                        6/25/2017</font></font></i></b><br>
                <b><i><font color="#FF0000"><font size="+0">Tom Hall</font></font></i></b>
                <br>
              </p>
              <p><a href="mailto:tomandhall@gmail.com">tomandhall@gmail.com
                </a><br> 
                <font color="#00FFFF"><font size="-1"><a
                      href="https://thalljiscience.github.io">Download</a></font></font></p>
            </center>
          </td>
          <td><a href="https://thalljiscience.github.io"><img src="bioIcon2.gif" border="0"
                height="146" width="320"></a>
            <center><font color="#00FFFF"><font size="-1"><a
                    href="https://thalljiscience.github.io">Back
                    to BioEdit Main Page</a></font></font></center>
          </td>
        </tr>
      </tbody>
    </table>
    <br>
    &nbsp;
    <br>
    <table style="width: 1176px; height: 12986px;" cellpadding="0"
      cellspacing="0">
      <caption><br>
      </caption><tbody>
      </tbody>
      <tbody>
        <tr>
          <td><img src="clear.gif" height="70" width="70"></td>
          <td><b><br><br></b>BioEdit has not been regularly
            maintained for several years.  The change log has not been updated since version 7.2.6.1. However, a few updates have
            been made simply to keep the program generally
            functional.&nbsp; The following list is not complete because
            the time has not been available to adequately keep
            track.&nbsp; The help file has not been updated to reflect
            most changes.<br><br>
            <b><font color="#000000">Since v7.2.6:</font></b>
            <ul>             
              <li>Rebuilt program to restore Ctrl+Right click functionality for reference sequence choice.</li>
            </ul> 
            <b><font color="#000000">Since v7.2.4:</font></b>
            <ul>             
              <li>Added functionality for https to fix download of sequence data from NCBI.</li>
            </ul> 
            <b><font color="#000000">Since v7.1.11:</font></b>
            <ul>             
              <li>The dot-plot feature for pairwise sequence comparison was brought back by request from a user (Under Sequence->Pairwise alignment->Dot Plot (pairwise comparison).</li>
              <li>Taxonomy mapping from sequence title was brought back under Sequence->Phylogeny / Taxonomy->Extract Taxonomy->(three mapping options).  Only bacterial and viral taxonomies are supported by default, but a custom table can be made by opening the file &lsaquo;BioEdit&rsaquo;\tables\ Bacterial_phylogeny.tab or &lsaquo;BioEdit&rsaquo;\tables\ Viral_Phylogeny.tab, copying the contents into MS Excel, and creating a custom table according to the same general template, then saving the custom file as a tab-delimited file.  If time permits, an auto-map of taxonomy from NCBI may be added in the future for convenience.</li>
            </ul>            
            <b><font color="#000000">Since v7.0.9:</font></b>
            <ul>
              <li>The pairwise alignment function was fixed to agree
                with examples in the help documentation.</li>
              <li>
                Right-click context menu added to title panel in
                sequence document.</li>
              <li>Support for local BLAST and accessory application
                launching fixed in Windows 7 64-bit version.</li>
              <li>GenBank sequence retrieval updated.<br>
              <li>String-to-floating point number conversion problems (hopefully) fixed for European computers with particular regional settings.<br>
              <li>Single-sequence translation in selected frames with codon usage reporting was fixed.  There was a memory bug when translated sequence was not exact multiple of three in length.<br>
              <li>Associating a phylogenetic tree with an alignment from the phylogenetic tree view was fixed.<br>
              <li>Six-frame translations were fixed to allow producing ORF lists from multiple sequences at once.<br>
              <li>Added option to save six-frame translations to a file rather than loading directly into text editor.<br>
              </li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.8:</font></b>
            <ul>
              <li>
                The restriction map utility was fixed.&nbsp; It was
                broken by the relocation
                of the BioEdit.ini file to the program install
                directory.</li>
              <li>
                An option was put into the preferences to allow
                temporary redirecting of
                file paths for blast database creation and search
                execution to the root
                directory.&nbsp; The reason for this is that if the name
                of the database
                directory (when doing a search) or source directory for
                a fasta file (when
                converting to a blast database) is too long, it can
                overload the maximum
                command line length for running blastall.exe or
                formatdb.exe (the NCBI
                command-line programs used for blast utilities).</li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.7:</font></b>
            <ul>
              <li>
                <font color="#000000">Moved BioEdit.ini to the program
                  current run directory.&nbsp;
                  Hopefully this will help on computers without write
                  access to the system
                  folder.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.5:</font></b>
            <ul>
              <li>
                <font color="#000000">Updated NCBI sequence retrieval to
                  deal with recent
                  changes at NCBI that broke the sequence retrieval
                  functions in the previous
                  version.</font></li>
              <li>
                <font color="#000000">Note:&nbsp; (not a change, per
                  se):&nbsp; Scoring
                  in pairwise alignment functions does not agree with
                  the example in the
                  help document.&nbsp; This will be investigated if/when
                  I have time.&nbsp;
                  I had considered removing the functions, but decided
                  no to on the grounds
                  that they generally work OK for quick manipulations
                  and can be useful as
                  a time-saving utility.&nbsp; They should not, however,
                  be used as a teaching
                  tool.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.4:</font></b>
            <ul>
              <li>
                <font color="#000000">Changing proxy setting for
                  sequence retrieval from
                  GenBank had no affect because the routine was looking
                  in the wrong directory
                  for the BioEdit.ini file.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.3:</font></b>
            <ul>
              <li>
                <font color="#000000">Bug in sequence retrieval from
                  GenBank fixed that
                  caused retrievals to come back saying " ... format
                  unrecognized".</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.2:</font></b>
            <ul>
              <li>
                <font color="#000000">A memory bug in RNA covariation
                  routine was fixed.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.1:</font></b>
            <ul>
              <li>
                <font color="#000000">The blast programs blastall and
                  formatdb were modified
                  to allow passing parameters in a file as well as on
                  the command line to
                  get around the Windows xommand line length limit.</font></li>
              <li>
                <font color="#000000">blastall and formatdb were
                  modified to allow passing
                  a darabase name with spaces (original verion tries to
                  parse the name into
                  multiple file names).</font></li>
              <li>
                <font color="#000000">When launching an alignment file
                  from the Windows
                  Explorer, BioEdit now tries to open the file in an
                  existing BioEdit instance
                  if there is one open, rather than opening a new one.</font></li>
              <li>
                <font color="#000000">Updated installer will hopefully
                  take care of the
                  trouble some people have been having installing
                  BioEdit.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v7.0.0:</font></b>
            <ul>
              <li>
                The optimal pairwise alignment routine was inadvertently
                overwritten with
                a test version that was only functional on nucleic acid
                sequences.&nbsp;
                This caused pairwise alignment on amino acid sequences
                to fail.&nbsp; The
                original alignment method was put back.</li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v6.0.7:</font></b>
            <ul>
              <li>
                Documentation (online help file) was updated for the
                first time since version
                5.0.6.&nbsp; However, my help compiler has a mild
                Turet's syndrome and
                sporadically tosses numbers in here and there, listing
                sentences by number
                that did not occur in lists in the original document
                (will have to be fixed
                later).</li>
              <li>
                Automatic links to NCBI BLAST via the World Wide Web
                were added back with
                current NCBI pages with editable addresses.</li>
              <li>
                Error on doing local blast searches when swap directory
                contained spaces
                fixed.&nbsp; However, BLAST won't work if entire command
                line is too long
                for Windows, so keeping the BioEdi t swap directory in a
                path name as short
                as possible (e.g. C:\BioEdit\Temp) is a good idea.</li>
              <li>
                Some context-sensitive help added (not complete).</li>
              <li>
                Fixed bug in graphic view that caused Error when view is
                plotted in dot
                conservation mode and the "reference" sequence is
                shorter than one of the
                alignment sequences.</li>
              <li>
                Fixed a bug in floating point conversion with
                non-English regional settings
                (when a ',' is used as decimal separator) that would
                cause an error when
                trying to translate nucleotide sequences into protein
                (bug manifested when
                reading in the "codon.tab" file and converting frequency
                numbers -- the
                regional settings were not being detected properly).</li>
              <li>
                Added support for mouse wheel in Alignment document
                window.&nbsp; Clicking
                the mouse wheel button toggles between vertical and
                horizontal scroll function.</li>
              <li>
                Fixed bug in right-click-based gap deletion in grouped
                sequences with group-locked
                alignment status (gaps were being deleted only for
                edited sequence and
                no entire locked group).</li>
              <li>
                A couple of other bugs relating to adding or deleting
                gaps around a residue
                with the right mouse button in a sequence which is a
                member of an alignment-locked
                grouped were fixed.&nbsp; Behavior was updated to only
                operate upon sequences
                that are not grouped together in an alignment-locked
                group with the sequence
                in question.</li>
              <li>
                Fixed bug in clustal auto-link that caused VERSION field
                to be lost in
                aligned sequences.</li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v6.0.6:</font></b>
            <br>
            <font color="#000000">This is a list of counterintuitive
              changes to
              deal with a set of counterintuitive features in the short
              term</font>
            <ul>
              <li>
                <font color="#000000">Removed automatic taxonomy mapping
                  support -- its
                  undocumented and caused confusion.</font></li>
              <li>
                <font color="#000000">Removed primer pair/oligo mapping
                  and Tm calculation
                  -- undocumented and caused confusion.</font></li>
              <li>
                <font color="#000000">Removed query functions for
                  sequence features --
                  again, undocumented and caused confusion.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v6.0.5:</font></b>
            <ul>
              <li>
                <font color="#000000">Removed "Auto-update View Options"
                  because it causes
                  long delays in Win2000 and XP.</font></li>
              <li>
                <font color="#000000">Fixed a bug in Graphic View new to
                  v6.0.5 causing
                  access violation if View Features mode is on in
                  document window.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.9</font></b>
            <ul>
              <li>
                Some memory troubles with grouped sequences fixed.&nbsp;
                Using sequence
                groups is now more stable.</li>
              <li>
                Access violations when deleting sequences in dot
                conservation mode or vertically
                scrolling very long alignments in dot conservation mode
                fixed.</li>
              <li>
                <font color="#000000">Retrieve sequences directly from
                  GenBank over the
                  WWW.</font></li>
              <li>
                Automatically retrieve GenBank info from GenBank for
                existing sequences
                for Genbank-fasta formatted titles (eg.
                gi|16082679|ref|NC_003132.1|).</li>
              <li>
                Automatically retrieve Pubmed references for GenBank
                sequences.</li>
              <li>
                Multiple levels of undo (up to 30).</li>
              <li>
                <span style="font-weight: bold;"></span>Restriction on
                number of sequences
                removed.&nbsp; All memory allocation for sequences was
                made dynamic.</li>
              <li>
                <span style="font-weight: bold;"></span>Paste over
                sequence blocks to quickly
                create subalignments based upon<span style="font-weight:
                  bold;">&nbsp;</span>a
                region of a larger alignment or produce alignment
                concatenations.&nbsp;</li>
              <li>
                <span style="font-weight: bold;"></span>Paste over
                titles, and paste onto
                (concatenate to) titles.</li>
              <li>
                <span style="font-weight: bold;"></span>Copy sequence
                titles added in a
                couple of variations.</li>
              <li>
                <span style="font-weight: bold;"></span>Copy nucleic
                acid sequence blocks
                in reverse complemented orientation</li>
              <li>
                <span style="font-weight: bold;"></span>Copy sequences
                in a tab-delimited
                format (for pasting into a spreadsheet)</li>
              <li>
                <span style="font-weight: bold;"></span>Find/replace
                function for sequence
                information</li>
              <li>
                <span style="font-weight: bold;"></span>Flexible
                find/replace in titles
                with indexed wildcards</li>
              <li>
                <span style="font-weight: bold;"></span>Filter leading
                and trailing characters
                off of sequence titles.</li>
              <li>
                <span style="font-weight: bold;"></span>Find titles
                equal to, beginning
                with, or containing lists of strings.</li>
              <li>
                <span style="font-weight: bold;"></span>Unique based
                upon sequences or
                title with a two-step sort + select &amp; delete process
                to remove redundant
                sequences or titles</li>
              <li>
                <span style="font-weight: bold;"></span>Query sequences
                by data by TITLE,
                LOCUS, DEFINITION, ACCESSION, VERSION / GI, PID (or
                SID), DBSOURCE, KEYWORDS,
                REFERENCES, FEATURES, COMMENT, and/or Phylogeny.</li>
              <li>
                <span style="font-weight: bold;"></span>Invert selection
                of sequences or
                residues</li>
              <li>
                <span style="font-weight: bold;"></span>Edit information
                for multiple sequences
                at once with Sequence--&gt;Edit<span style="font-weight:
                  bold;">&nbsp;</span>all
                selected.</li>
              <li>
                <span style="font-weight: bold;"></span>Extract
                subsequences (including
                orientation for DNA) from lists of<span
                  style="font-weight: bold;">&nbsp;</span>coordinates
                (with choice of alignment-gapped or true
                positions)&nbsp;</li>
              <li>
                <span style="font-weight: bold;"></span>Query and
                extract annotated sequence
                features.</li>
              <li>
                <span style="font-weight: bold;"></span>Map true
                positions from lists of
                alignment positions.</li>
              <li>
                <span style="font-weight: bold;"></span>Add tabulated
                taxonomy information
                to sequences.</li>
              <li>
                <span style="font-weight: bold;"></span>Several variable
                ways of renaming
                titles in batch mode (Sequence--&gt;Rename--&gt;...)</li>
              <li>
                <span style="font-weight: bold;"></span>Several ways to
                filter out sequences
                based upon inclusion/exclusion<span style="font-weight:
                  bold;">&nbsp;</span>of
                specific characters, length, or number of leading,
                trailing, or total gaps.</li>
              <li>
                <span style="font-weight: bold;"></span>Sort by
                phylogeny</li>
              <li>
                <span style="font-weight: bold;"></span>Invert current
                sequence order</li>
              <li>
                <span style="font-weight: bold;"></span>Map oligos and
                primer pairs to
                alignments based upon mismatch threshold</li>
              <li>
                <span style="font-weight: bold;"></span>Calculate DNA
                oligo Tm's with mismatches
                based upon thermodynamic<span style="font-weight: bold;">&nbsp;</span>parameters
from
                John SantaLucia's lab.</li>
              <li>
                <span style="font-weight: bold;"></span>Calculate Tm's
                for oligos and compare
                them to same region of other&nbsp;<span
                  style="font-weight: bold;"></span>sequences
                from alignment window using right mouse button clicks.</li>
              <li>
                <span style="font-weight: bold;"></span>Several
                operations on local sequence
                region selection with right&nbsp;<span
                  style="font-weight: bold;"></span>mouse
                button click.</li>
              <li>
                <span style="font-weight: bold;"></span>Speed enhanced
                on pairwise alignment
                of large similar sequences</li>
              <li>
                <span style="font-weight: bold;"></span>Align sequences
                to an existing
                alignemnt by pairwise alignment to<span
                  style="font-weight: bold;">&nbsp;</span>phylogenetic
                near neighbor in existing alignment.</li>
              <li>
                <span style="font-weight: bold;"></span>Choose
                individual restriction enzymes
                when creating a restriction map.</li>
              <li>
                Easily flush an alignment out with gaps to produce a
                rectangular matrix</li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.8:</font></b>
            <ul>
              <li>
                <font color="#000000">Made a minor adjustment to
                  preferences for characters
                  to be treated as residues -- There's an option to
                  either include them as
                  positions or ignore them completely for similarity/ID
                  shading.</font></li>
              <li>
                <font color="#000000">Fixed a bug in pasting into the
                  single sequence editor
                  window.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.7:</font></b>
            <ul>
              <li>
                <font color="#000000">Protein alignments in new
                  installations may have
                  shading problems because when the option to explicitly
                  specify all characters
                  to be considered valid for calculations (including
                  ID/similarity shading),
                  'N' was accidentally missed for amino acids.&nbsp;
                  Several installations&nbsp;
                  probably have gone out with 'N' missing from the list
                  of amino acids.&nbsp;
                  This can cause protein alignments to have
                  similarity/identity shading errors.&nbsp;
                  'N' has been put back into the default.</font></li>
              <li>
                <font color="#000000">Access violations could occurr
                  when typing in an
                  RNA structure mask while in RNA base pair coloring
                  mode, and when there
                  is a non-symmetric number of openeing vs. closing
                  brackets.&nbsp; This
                  has been fixed.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.6:</font></b>
            <ul>
              <li>
                <font color="#000000">Added base pair mask support for
                  RNA alignments.</font></li>
              <li>
                <font color="#000000">Added base-pair coloring for RNA
                  alignments with
                  user-defined color scheme.</font></li>
              <li>
                <font color="#000000">Fixed bug in creating local Blast
                  databases from
                  within a sequence document.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.5:</font></b>
            <ul>
              <li>
                <font color="#000000">Added support for phylogenetic
                  trees without distance
                  information.</font></li>
              <li>
                <font color="#000000">Added save function for
                  phylogenetic trees.</font></li>
              <li>
                <font color="#000000">Added linking of phylogentic trees
                  to an alignment.</font></li>
              <li>
                <font color="#000000">Multiple trees may be imported
                  from one or several
                  files.</font></li>
              <li>
                <font color="#000000">Incorporated phylogentic trees are
                  saved with alignment
                  in BioEdit alignment format.</font></li>
              <li>
                <font color="#000000">Added option to view trees with or
                  without distance
                  information.</font></li>
              <li>
                <font color="#000000">Updated sequence label editing to
                  allow spaces in
                  comments.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.4:</font></b>
            <ul>
              <li>
                <font color="#000000">Added node flipping to
                  phylogenetic tree viewer.</font></li>
              <li>
                <font color="#000000">Added printing to phylogenetic
                  tree viewer.</font></li>
              <li>
                <font color="#000000">Fixed bug in the "append
                  alignments" (the first alignment
                  was appended to itself) -- bugged version was only out
                  for one day.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.3:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed some issues with accessory
                  app linking (problems
                  keep creeping up related to the change made in version
                  5.0.0 to specify
                  a swap folder for temp files).</font></li>
              <li>
                <font color="#000000">Added the option to merge
                  alignments by appending
                  one alignment to the end of another.</font></li>
              <br>
              <font color="#000000">Added a ruimentary tree viewer (VERY
                unfinished
                -- only reads phylip trees, only in "phylogram" view, no
                distance ruler
                yet, no print, save or manipulate yet) -- The reason for
                having an internal
                tree viewer is to in the future allow flipping of nodes
                and resaving the
                tree for use in nicer tree viewing programs like
                TreeView that allow better
                formatting, but don't have manipulation options.&nbsp;
                Also, it would be
                nice in the future to directly link phylogenetic tree
                views to alignment
                files for quick phylogeny reference.</font>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.2:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed a bug (hopefully) that
                  caused accessory apps
                  to not run correctly (bad command or filename error).</font></li>
              <li>
                <font color="#000000">Added column anchoring to protect
                  areas that are
                  already aligned.</font></li>
              <li>
                <font color="#000000">Updated translation toggling so
                  that gaps that are
                  expanded back from a single gap (when toggling from a
                  protein view to a
                  nucleotide view) are always "locked" (to avoid
                  problems with inadvertently
                  crunching off-screen gaps that spuriously toggled back
                  as "unlocked" when
                  aligning by hand).</font></li>
              <li>
                <font color="#000000">Fixed a bug that caused RNA
                  covariation, potential
                  pairings, or mutual information analyses to crash when
                  "mask numbering"
                  was chosen but there was no numbering mask specified.</font></li>
              <li>
                <font color="#000000">Fixed newly introduced bug the
                  caused MI Examiner
                  to crash when no masks are specified.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.1:</font></b>
            <ul>
              <li>
                <font color="#000000">Updated Mutual Information
                  Examiner to handle lists
                  of positions.</font></li>
              <li>
                <font color="#000000">Added several options to Mutual
                  Information preferences
                  to allow variable display of list information in MI
                  Examiner.</font></li>
              <li>
                <font color="#000000">Added option to export and import
                  color table files.</font></li>
              <li>
                <font color="#000000">Fixed a bug that caused color
                  table to lock sometimes.</font></li>
              <li>
                <font color="#000000">Added option to toggle
                  translations using the current
                  selected residue as a reference frame.</font></li>
              <li>
                <font color="#000000">Fixed a bug that caused document
                  to appear hung when
                  sort by base frequency was chosen without a selected
                  column.</font></li>
              <li>
                <font color="#000000">Added ability to sort by selected
                  range of residues.</font></li>
              <li>
                <font color="#000000">No longer append sequence lengths
                  to titles when
                  saving Fasta files.</font></li>
              <li>
                <font color="#000000">Added preference (on "General"
                  tab) to require verification
                  for revert to saved function.</font></li>
              <li>
                <font color="#000000">Fixed problem with creating Local
                  Blast databases
                  which would occur on machines with no NCBI tools
                  installed.&nbsp;</font></li>
              <li>
                <font color="#000000">Accessory apps and Blast Internet
                  client are now
                  launched through an auto-generated batch file.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v5.0.0:</font></b>
            <ul>
              <li>
                <font color="#000000">Added the option to specify
                  user-defined temp (swap)
                  directory for all temporary files and user-defined
                  BLAST database directory
                  so that BioEdit may be used from a drive space that
                  does not allow writing
                  (actually added to v5.0.0 a few days before 5.0.1 came
                  out).</font></li>
              <li>
                <font color="#000000">Fixed some bugs associated with
                  running accessory
                  apps from version 5.0.0 after the temp directory
                  option was added.<b>&nbsp;</b></font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.10:</font></b>
            <br>
            <font color="#000000">(Probably missed a few -- I hadn't
              updated this
              file for a while)</font>
            <ul>
              <li>
                <font color="#000000">Added options to specify which
                  characters in nucleic
                  acid and amino acid sequences are considered to be
                  valid for calculations.</font></li>
              <li>
                <font color="#000000">Added options for sorting
                  sequences by title, LOCUS,
                  DEFINITION, ACCESSION, PID/NID, REFERENCES, COMMENTS
                  and residue frequency
                  in a selected column.</font></li>
              <li>
                <font color="#000000">Added options to annotate
                  sequences with graphical
                  features, using the standard GenBank FEATURES tags as
                  the standard for
                  keeping track of annotation "type", but allowing full
                  user control over
                  feature names,&nbsp; descriptions, colors, shapes and
                  positions.</font></li>
              <li>
                <font color="#000000">Added function to automatically
                  annotate graphical
                  features using existing GenBank FEATURES field</font></li>
              <li>
                <font color="#000000">Added options for grouping
                  sequences into groups
                  or families, with user-defined coloring of grouped
                  titles and the option
                  to lock the alignment of grouped sequences to allow
                  synchronized hand alignment&nbsp;
                  adjustments between grouped and locked sequences.</font></li>
              <li>
                <font color="#000000">Added option to lock individual
                  sequences to prevent
                  accidental edits.</font></li>
              <li>
                <font color="#000000">Updated the BioEdit file format to
                  save graphical
                  annotations, sequence group information, locked
                  sequence information and
                  positional flags.</font></li>
              <li>
                <font color="#000000">Added function to update
                  previously saved BioEdit
                  plasmid maps with a new restriction enzyme file when
                  the ReBase file is
                  changed (changing the enzyme.tab file can cause
                  mis-indexing of restriction
                  enzyme sites in existing plasmid files).</font></li>
              <li>
                <font color="#000000">Updated shaded graphic view to
                  correctly do conservation
                  (dot) view when that view mode is selected in the
                  alignment window (no
                  to create a new dot-plotted alignment and any sequence
                  can be used as the
                  reference rather than just the top sequence).</font></li>
              <li>
                <font color="#000000">Added a "Overwrite" and "Insert"
                  modes to both on-screen
                  and single-sequence editor sequence editing.</font></li>
              <li>
                <font color="#000000">Updated local BLAST interface to
                  feed multiple sequence
                  automatically for a batch job, and to allow
                  specification of a permanent
                  file output and the option to open output
                  automatically in the BioEdit
                  text editor or not.&nbsp;</font></li>
              <li>
                <font color="#000000">Modified some menus (Edit and
                  Sequence menus, mainly).&nbsp;
                  The same options (and more) are still there, but
                  submenus having been created
                  to group certain related functions including:
                  gap-realted functions, pairwise
                  alignment functions, simple manipulations (reverse,
                  uppercase, lowercase,
                  remove numbers), and search functions.</font></li>
              <li>
                <font color="#000000">The "No BLAST" version (BioEdit
                  without local BLAST
                  packaged) was done away with -- it's too much of&nbsp;
                  pain to duplicate
                  everything but one extra function and make a whole
                  independent installation
                  when relatively few people download that one anyway).</font></li>
              <li>
                <font color="#000000">At the last minute -- added an
                  option to verbally
                  read back sequences (protein, nucleic acid, or
                  anything else) to verify
                  hand-typed sequence entries.</font></li>
              <li>
                <font color="#000000">The docuentation was updated
                  12/18/2000.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.8.9:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed the (re-introduced) bug that
                  caused floating
                  point number formatting error when translating nucleic
                  acid sequences on
                  non-US regional settings machines</font></li>
              <br>
              <font color="#000000">Fixed a couple of bugs that affected
                reading
                in of sample name and version on ABI 310 files</font>
            </ul>
            <b><font color="#000000">v4.8.8 -- forgot to document
                changes</font></b>
            <p><b><font color="#000000">Bug fixes/changes since v4.8.7:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Removed the internal web browser</font></li>
              <li>
                <font color="#000000">Fixed a bug in codon frequency
                  reporting (nucleotide
                  translations) that caused floating point numbers to
                  sometimes get jumbled
                  up in the output</font></li>
              <br>
              <font color="#000000">Added a conservation plot toggle to
                the main
                alignment window with the option to choose the character
                to draw for a
                "conserved residue" (the "reference" sequence defaults
                to the first sequence
                in an alignment -- RIGHT-CLICK the mouse over any
                sequence to change it
                to the current reference sequence for this view option)</font>
              <li>
                <font color="#000000">Fixed a minor bug in setting of
                  custom menu shortcuts
                  for alignment documents.&nbsp; The "Import from
                  clipboard" should now work
                  with a user-defined menu shortcut.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.6:</font></b>
            <ul>
              <li>
                <font color="#000000">Added optimal pairwise alignment
                  routine with a summary
                  of percent identity and similarity (multiple choices
                  for scoring matrix
                  for amino acid alignment, user-defined gap initiate,
                  gap extend and nucleotide
                  match parameters, and option for global alignment or
                  alignment allowing
                  ends to slide over each other).</font></li>
              <li>
                <font color="#000000">Added routine to create an
                  alignment that plots all
                  residues that are identical to the top sequence in an
                  alignment as a dot.</font></li>
              <li>
                <font color="#000000">Added a cheap version of a dot
                  plotter for plotting
                  matrix similarity between two sequences -- total
                  sequence length product
                  cannot exceed 2000 x 2000, however, so it's not useful
                  for genome comparisons
                  or even moderately large sequences.</font></li>
              <li>
                <font color="#000000">Added option to calculate the
                  similarity and/or identity
                  between two sequences as they are aligned.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.5:</font></b>
            <ul>
              <li>
                <font color="#000000">3/10/2000 -- added NCBI blastcl3
                  to distribution
                  and added a more versatile linking interface for the
                  BLAST client -- the
                  old BLAST client 2 has apparently been disabled</font></li>
              <li>
                <font color="#000000">3/10/2000 -- Added block
                  convention to user-defined
                  motif search.&nbsp; [] brackets indicate ambiguous
                  residues and {} brackets
                  indicate exclusions (eg [LIV] will accept 'L', 'I' or
                  'V', whereas {LIV}
                  will accept anything BUT 'L', 'I' or 'V'.</font></li>
              <li>
                <font color="#000000">3/10/2000 -- Fixed bug that caused
                  an eror and a
                  blank doc to be opened when the alignment control bar
                  was hidden</font></li>
              <li>
                <font color="#000000">3/10/2000 -- Fixed a couple of
                  bugs related to reading
                  and writing of GenBank files (lines beginning with
                  numbers were truncated
                  and manually updating fields sometimes resulted in
                  misformatting)&nbsp;</font></li>
              <li>
                <font color="#000000">Added an alignment merge funcion
                  which merges two
                  alignments based upon a standard common to both
                  alignments&nbsp;</font></li>
              <li>
                <font color="#000000">Added batch export of ABI raw
                  trace data.</font></li>
              <li>
                <font color="#000000">Fixed the recently introduced
                  disabling of the ABI/SCF
                  trace print option</font></li>
              <li>
                <font color="#000000">Fixed metafile copy and print of
                  reverse-complemented
                  traces</font></li>
              <li>
                <font color="#000000">Improved default scaling of SCF
                  traces for printing
                  and metafile copy</font></li>
              <li>
                <font color="#000000">Fixed bug in graphic view which
                  could affect first
                  time use of the graphic view when used with non-US
                  Windows regional settings&nbsp;</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.4:</font></b>
            <ul>
              <li>
                <font color="#000000">1/28/2000 -- Fixed bug that came
                  out when creating
                  a mask (must have been introduced in version
                  4.7.8).&nbsp; This bug caused
                  two masks to be formed and sequence memory for the
                  first two sequences
                  to get cross-linked. (caused by accidentally
                  re-assigning a memory pointer
                  rather than an entire data structure).</font></li>
              <li>
                <font color="#000000">1/27/2000 -- Fixed a bug that was
                  introduced with
                  the very last bug fix concerning ClustalW 1.4 to 1.8
                  upgrading.&nbsp; With
                  ClustalW 1.4 (the version distributed with BioEdit)
                  the sequence names
                  would get mangled and GenBank information was not
                  being copied into the
                  new alignment.&nbsp; This should be fixed in version
                  4.8.5 installations
                  downloaded after 1/27/2000</font></li>
              <li>
                <font color="#000000">Fixed potential bug that comes out
                  if you replace
                  the clustalw 1.4 (the version distributed with
                  BioEdit) with clustalw 1.8
                  in the apps folder -- clustalw 1.8 will rearrange the
                  order of the input
                  sequences, causing sequences and titles to become
                  scrambled in the new
                  alignment -- this has been fixed.</font></li>
              <li>
                <font color="#000000">Added raw data view and analyzed
                  and raw data export
                  from .abi trace files.</font></li>
              <li>
                <font color="#000000">Added option to change starting
                  point of ruler bar
                  in alignment window.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.3:</font></b>
            <ul>
              <li>
                <font color="#000000">1/4/2000 -- Fixed bug in read-in
                  filter for Phlyip
                  3.2 (non-interleaved) files that cauesed DNA/RNA
                  alignments to be detected
                  as protein</font></li>
              <li>
                <font color="#000000">2/20/1999 -- Added helical wheel
                  diagrams for proteins.&nbsp;
                  Select a protein or portion of a protein and choose
                  Sequence-&gt;Protein-&gt;Helical
                  Wheel Diagram</font></li>
              <li>
                <font color="#000000">Fixed bug in accessory app
                  launching that causes
                  spaces to be lost in command prefixes for inputs and
                  checkboxes</font></li>
              <li>
                <font color="#000000">Added option to accessory app
                  launching to auto-feed
                  sequences in separate files (eg, for Lalign from the
                  Fasta package)</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.2:</font></b>
            <ul>
              <li>
                <font color="#000000">12/13/1999 -- Fixed bug in
                  hydrophobic moment profiles
                  -- had an exponent missing in Fourier equation&nbsp;</font></li>
              <li>
                <font color="#000000">12/13/1999 -- Added hydrophobic
                  moment matrix plots
                  (0-180 degrees)</font></li>
              <li>
                <font color="#000000">12/13/1999 -- fixed a couple of
                  bugs in trace viewer
                  that caused problems with non-U.S. regional settings</font></li>
              <li>
                <font color="#000000">12/10/1999 -- Added support for
                  SCF version 2.00
                  files</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.1:</font></b>
            <ul>
              <li>
                <font color="#000000">Added options to accessory
                  application interface:</font></li>
              <ol>
                <li>
                  <font color="#000000">Bypass interface when running an
                    accessory</font></li>
                <li>
                  <font color="#000000">Link output(s) of one accessory
                    to another/other
                    accessory(s)</font></li>
                <li>
                  <font color="#000000">Create accessory app links that
                    are only called by
                    other accessory links and do not show up on
                    accessory app menu</font></li>
                <li>
                  <font color="#000000">Automatically truncate titles to
                    10 characters for
                    programs that don't accept long names and/or change
                    spaces to '_'</font></li>
                <li>
                  <font color="#000000">Rename redundant sequence names
                    and generate a report
                    file specifying title changes</font></li>
              </ol>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.8.0:</font></b>
            <ul>
              <li>
                <font color="#000000">12/2/1999 -- Fixed a bug in
                  accessory application
                  launching that could cause a memory error when running
                  an app from a window
                  that is not a sequence alignment document</font></li>
              <li>
                <font color="#000000">12/1/1999 -- Fixed on-screen title
                  editing so that
                  double-clicking on a selected title still opens the
                  sequence editing box
                  and overrides title editing</font></li>
              <li>
                <font color="#000000">12/1/1999 -- Added option to
                  disable on-screen editing</font></li>
              <li>
                <font color="#000000">12/1/1999 -- Fixed bug in
                  consensus sequence generation.
                  A consensus generated from less than all of the
                  sequences in an alignment
                  was using frequencies calculated using the total
                  number of sequences rather
                  than the selected number</font></li>
              <br>
              <font color="#000000">12/1/1999 -- Fixed bug that caused
                an access
                violation when re-activating a document that had
                sequences deleted (introduced
                with version 4.7.8)</font>
              <li>
                <font color="#000000">Fixed a very bad bug that was
                  introduced in version
                  4.7.8.&nbsp; Pasting sequences within an alignment
                  document caused sequence
                  structures to become cross-linked due to a duplicated
                  pointer.&nbsp; This
                  would lead to memory access violations and possible
                  loss of data.&nbsp;
                  This has been fixed.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.9:</font></b>
            <ul>
              <li>
                <font color="#000000">Added support for SCF sequence
                  trace files</font></li>
              <li>
                <font color="#000000">Added horizontal and vertical zoom
                  controls to sequence
                  trace form</font></li>
              <li>
                <font color="#000000">Added crosshairs with trace
                  position reporting in
                  trace form</font></li>
              <li>
                <font color="#000000">Added batch conversion of ABI
                  files to SCF format</font></li>
              <li>
                <font color="#000000">Traces draw faster on the screen</font></li>
              <li>
                <font color="#000000">Fixed another bug in on-screen
                  sequence title editing
                  (when vertical bar was scrolled, it was causing
                  trouble with the identity
                  of the edited title)</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.8:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed bug in local BLAST linking
                  that caused failure
                  when operating from a directory with a space in its
                  name and using a mtrix
                  other than Blosum62</font></li>
              <li>
                <font color="#000000">Fixed a similar bug in linking to
                  ReadSeq for importing
                  and exporting of extra file formats</font></li>
              <li>
                <font color="#000000">Added "Import from Clipboard" and
                  "New from Clipboard"
                  functions unde the file menu -- Clipboard contents are
                  treated exactly
                  like a file an automatic linking to ReadSeq for
                  importing formats that
                  BioEdit does not internally read is maintained</font></li>
              <li>
                <font color="#000000">Added function for inserting
                  multiple gaps at selected
                  residue positions or at specified position in selected
                  sequences</font></li>
              <li>
                <font color="#000000">Updated on-screen title editing --
                  use the arrow
                  keys to scroll up and down when changing several
                  titles in the same document</font></li>
              <li>
                <font color="#000000">Fixed a bug in plasmid drawing
                  that caused a floating
                  point error when adding a very short feature to a
                  circular plasmid</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.7:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed a bug that caused keystrokes
                  to appear in sequences
                  while in edit mode but changing sequence titles or
                  accessing menus with
                  alt+ keystrokes</font></li>
              <li>
                <font color="#000000">User-defined motif search function
                  added for amino
                  acid, nucleic acid, or simple text.</font></li>
              <li>
                <font color="#000000">Memory allocation for sequences
                  changed to use less
                  memory (memory was being allocated to set aside space
                  for 20,000 sequence
                  structures regardless of the number of sequences in a
                  file -- now a single
                  pointer is set aside for the max number of allowed
                  sequences (20,000),
                  but no memory is allocated until a sequence is
                  created)</font></li>
              <li>
                <font color="#000000">Bug in ABI trace reader was fixed
                  that caused trace
                  / nucleotide colors to be mismatched when the file
                  contained an order other
                  than "GATC"</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.6:</font></b>
            <br>
            &nbsp;
            <ul>
              <li>
                <font color="#000000">ABI files may be edited and saved,
                  and edited sequence
                  may be reverted to the original, non-editable
                  version.&nbsp; The edited
                  sequence is opened as a sequence/alignment doc upon
                  opening an ABI file.</font></li>
              <li>
                <font color="#000000">Extended capabilities of
                  restriction mapping module.&nbsp;
                  Long sequences may be mapped (tested up to 1.75 Mb on
                  machine with 128
                  Mb RAM).&nbsp; My mapping procedure is pretty slow,
                  but 500,000 bp runs
                  in ~2 minutes on a 366 MHz Celeron).&nbsp; Progress
                  indicators added for
                  long sequences.</font></li>
              <li>
                <font color="#000000">Added local BLAST interface to
                  main program interface
                  (rather than just in sequence/alignment docs)</font></li>
              <li>
                <font color="#000000">The paste in the sequence edit box
                  that caused it
                  to go very slow with long sequences has been fixed
                  (&gt;500,000 residue pasting
                  is no longer a problem).</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.5:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed link to readseq for
                  import/export of extra
                  formats that made it fail with long file names</font></li>
              <li>
                <font color="#000000">Added read-in filter for GCG
                  format</font></li>
              <li>
                <font color="#000000">a couple of other minor bug fixes</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.4:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed a few leftover bugs in the
                  graphic shaded views
                  (mostly spacing)</font></li>
              <li>
                <font color="#000000">user control over vertical spacing
                  of sequences</font></li>
              <li>
                <font color="#000000">Added view of similarity scoring
                  matrix files from
                  doc windows</font></li>
              <li>
                <font color="#000000">Edit titles on screen (highlight a
                  title, then click
                  it again)</font></li>
              <li>
                <font color="#000000">Fixed bug in local blastp --
                  matrix choices other
                  than BLOSUM62 should work now</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.3:</font></b>
            <ul>
              <li>
                <font color="#000000">A rich text export was added to
                  the shaded graphic
                  view utility.&nbsp; This uses the highlight command in
                  the Rich Text Format
                  specification 1.5, and is probably compatable only
                  with relatively modern
                  word-processors.</font></li>
              <li>
                <font color="#000000">Made a few changes and fixes to
                  the shaded view.&nbsp;</font></li>
              <li>
                <font color="#000000">Added a "find in next title"
                  function to the main
                  document window</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.2:</font></b>
            <ul>
              <li>
                <font color="#000000">Clustal format (.aln) filter was
                  updated so it now
                  compatible with the sequence numbers option in
                  ClustalX</font></li>
              <li>
                <font color="#000000">A "delete numbers" option was put
                  in under the sequence
                  menu to allow easier compatability with copying and
                  pasting from numbered
                  formats</font></li>
              <li>
                <font color="#000000">In the shaded graphic view, an
                  option to print lines
                  continuously (not broken by a space every ten
                  residues) was added</font></li>
              <li>
                <font color="#000000">An option to place translations
                  below nucleotide
                  sequences in either one-letter or three-letter format
                  was added (allows
                  for coloring with amino acid color table colors, but
                  no similarity shading
                  of the amino acids in this view).</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.1:</font></b>
            <ul>
              <li>
                <font color="#000000">8/27/1999: Added option for
                  outlines around identity
                  and similarity shading in shaded graphic view of
                  alignments</font></li>
              <li>
                <font color="#000000">8/27/1999:&nbsp; Fixed small bug
                  in restriction maps.&nbsp;
                  If more than one enzyme cut at the very position
                  starting a new line on
                  the printed map, only the first enzyme would show on
                  the map (though all
                  would appear in the lists).<b>&nbsp;</b></font></li>
              <li>
                <font color="#000000">User options are provided for a
                  (hopefully) more
                  useful consensus sequence generation.</font></li>
              <li>
                <font color="#000000">The ABI trace viewer was updated
                  to provide summary
                  information and to allow copying of the trace window
                  as a bitmap (in last
                  version, I forgot to add that the option to copy print
                  pages as Metfiles
                  was also added).</font></li>
              <li>
                <font color="#000000">Support for reading Clustal format
                  has been added
                  (no save, only open).</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.7.0:</font></b>
            <ul>
              <li>
                <font color="#000000">The graphic shaded view was
                  improved.&nbsp; It's
                  more like a print preview.&nbsp; Also, the option to
                  add numbers to the
                  sides of the sequences has been added</font></li>
              <li>
                <font color="#000000">*Another* introduced bug was fixed
                  in the accessory
                  applications launch interface.&nbsp; This bug was
                  preventing input and
                  output file name command prefixes from being added to
                  the command line
                  (needless to say, many apps would not run).</font></li>
              <li>
                <font color="#000000">Font size control has been added
                  to the single sequence
                  editing box.&nbsp;</font></li>
              <li>
                <font color="#000000">Accessory apps no longer need to
                  be run from an alignment
                  doc unless they operate upon sequence output.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.6.3:</font></b>
            <ul>
              <li>
                <font color="#000000">Added support for ABI
                  chromatograms.&nbsp; Display
                  traces, import sequence, select, copy, export, reverse
                  complement and print
                  with professional-looking output.</font></li>
              <li>
                <font color="#000000">Fixed a couple of memory leaks
                  (some sequence memory
                  was not being released and some memory allocated
                  during manipulation of
                  plasmid images and matrix plot images was "leaking")</font></li>
              <li>
                <font color="#000000">Fixed a bug in formatting output
                  of GenBank files
                  containig extra field data (REFERENCES, SOURCE, etc)</font></li>
              <li>
                <font color="#000000">7/20/1999:&nbsp; Fixed bug in
                  translations with codon
                  usage that occurred when sequences had characters
                  other than A, G, C, T
                  or U.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.6.2:</font></b>
            <ul>
              <li>
                <font color="#000000">7/6/1999 -- corrected error in
                  molecular mass calculations
                  for nucleic acids (there was an extra water molecule
                  (18 Da) for every
                  internal nucleotide)</font></li>
              <li>
                <font color="#000000">Corrected 'Del' key behavior in
                  edit residues mode
                  and in single sequence edit interface</font></li>
              <li>
                <font color="#000000">Added sequence number to beginning
                  of mouse position
                  labels</font></li>
              <li>
                <font color="#000000">Fixed alignment of ruler ticks for
                  non-type-set fonts</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.6.1:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed a bug that caused
                  simultaneous entry of characters
                  into mutual information examiner X or Y entry and
                  sequence when in Edit
                  Residues mode</font></li>
              <li>
                <font color="#000000">Fixed glich that caused ruler to
                  not extend to end
                  of window in very high screen resolutions with small
                  font size</font></li>
              <li>
                <font color="#000000">Fixed problem (I think) with
                  disappearing window
                  controls (minimize, maximize and close controls)</font></li>
              <li>
                <font color="#000000">Fixed bug in accessory application
                  interface that
                  caused spaces between input and output file prefixes
                  and filenames to not
                  be added to the command line&nbsp;</font></li>
              <li>
                <font color="#000000">Modified accessory application
                  interface to use short
                  path names for input files (Some programs would not
                  run correctly from
                  directories with spaces in them, eg Program Files)</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.10:</font></b>
            <ul>
              <li>
                <font color="#000000">Scroll a full screen at a time
                  with the PageUp/PageDown
                  keys and with Ctrl+arrow keys -- also refined cursor
                  movement in edit mode
                  and keyboard-controlled scrolling</font></li>
              <li>
                <font color="#000000">BioEdit now remembers your last
                  window state as the
                  default</font></li>
              <li>
                <font color="#000000">Fixed a bug in the single sequence
                  edit window --
                  applying twice in a row after creating a new sequence
                  no longer causes
                  an access violation</font></li>
              <li>
                <font color="#000000">Improved speed of dynamic shading
                  of protein alignments
                  in the edit window&nbsp;</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.9:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed bug in edit-window shading
                  (maximizing window
                  shouldn't cause a problem now).</font></li>
              <li>
                <font color="#000000">Added Select to End, Select to
                  Beginning and Select
                  Residues of&nbsp; Selected Sequences functions&nbsp;</font></li>
              <li>
                <font color="#000000">Changed open and save file dialogs
                  to remember file
                  extensions and last directory as defaults</font></li>
              <li>
                <font color="#000000">Added recent file list to the file
                  menu</font></li>
              <li>
                <font color="#000000">Added function to open an edit box
                  at the current
                  cursor position (when in Edit Residues mode)&nbsp;</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.8:</font></b>
            <ul>
              <li>
                <font color="#000000">Added identity/similarity-based
                  shading in the alignment
                  window</font></li>
              <li>
                <font color="#000000">Fixed a bug in the matrix plotter
                  which caused it
                  to periodically crash when plotting a half matrix
                  (above diagonal only).</font></li>
              <li>
                <font color="#000000">Fixed bug in N-best reporting of
                  mutual information
                  and N-best and P-best reports when data are organized
                  for the whole alignment
                  rather than by position (I forgot that those things
                  had never been finished).</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.7:</font></b>
            <ul>
              <li>
                <font color="#000000">Added information-based searching
                  for conserved segments
                  (may be useful for designing PCR primers or defining
                  motifs)</font></li>
              <li>
                <font color="#000000">Made the sequence titles window
                  adjustable in width</font></li>
              <li>
                <font color="#000000">Fixed bug in six-frame
                  translations:&nbsp; An "N"
                  in the sequence no longer causes an amino acid to be
                  skipped</font></li>
              <li>
                <font color="#000000">Added a "Minimize Alignment to
                  Mask" function to
                  allow trimming an alignment to positions present in a
                  mask sequence</font></li>
              <li>
                <font color="#000000">Added positional numerical summary
                  for nucleotide
                  alignments</font></li>
              <li>
                <font color="#000000">Changed entropy plots to area
                  plots that resize with
                  the window</font></li>
              <li>
                <font color="#000000">Added a full-text summary of
                  entropy (Hx) values
                  when an entropy plot is run&nbsp;</font></li>
              <li>
                <font color="#000000">Got rid of the "three-way
                  information" thing (too
                  much noise to be useful)</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.6:</font></b>
            <ul>
              <li>
                <font color="#000000">The vertical split has been
                  improved -- no longer
                  flickers wildly when going between windows (also, no
                  longer "left" and
                  "right" panes -- it's just one window now).&nbsp; The
                  horizontal split
                  view is still not-so-great.</font></li>
              <li>
                <font color="#000000">Added option to hide panels for
                  controls at top (to
                  allow for a bigger sequence editing window)</font></li>
              <li>
                <font color="#000000">Sped-up right-click driven
                  insertions and deletions.&nbsp;
                  Added option to reverse mouse buttons (to allow
                  inserting and deleting
                  gaps with a left click)</font></li>
              <li>
                <font color="#000000">Fixed glich in window updating
                  when switching between
                  color and B&amp;W</font></li>
              <li>
                <font color="#000000">Fixed glich in translations on
                  restriction maps --
                  spacing was off if an "X" resulted from a codon with a
                  non-A, C, G or U/T
                  character.</font></li>
              <li>
                <font color="#000000">Added menu items for all
                  button-driven controls</font></li>
              <li>
                <font color="#000000">"Lock gaps" no longer causes an
                  error when residue
                  selection goes beyond the end of a sequence.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.5:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed bug in residue selecting
                  that was introduced
                  with the sequence handling rewrite:&nbsp; The location
                  of a selected block
                  of residues was not scrolling correctly in the
                  horizontal direction</font></li>
              <li>
                <font color="#000000">Changed the way the matrix plotter
                  draws matrices.&nbsp;
                  The speed is improved several hundred fold.&nbsp;</font></li>
              <li>
                <font color="#000000">Added threshold control to matrix
                  plotter.</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.4:</font></b>
            <ul>
              <li>
                <font color="#000000">Fixed glich in amino acid
                  composition reporting --
                  previously, non-amino acid characters were assigned a
                  mass of 110 Da.&nbsp;
                  Now, only 'X' is assigned 110, other characters are 0
                  Da (not counted as
                  amino acids).&nbsp; The weight of 'B' is considered
                  the average of 'D'
                  and 'N', and the weight of 'Z' is the average of 'E'
                  and 'Q'.&nbsp; Also,
                  a stop codon at the end of a sequence no longer adds 1
                  amino acid to the
                  reported length of the sequence.</font></li>
              <li>
                <font color="#000000">A couple of changes were made to
                  the accessory application
                  configuration interface and the way accessory apps are
                  run:</font></li>
              <ol>
                <li>
                  <font color="#000000">Options to include or not
                    include the input and output
                    files in the command line were added, as well as the
                    option to include
                    them at the beginning or end of the command line.</font></li>
                <li>
                  <font color="#000000">Changed the method of calling
                    accessory apps:&nbsp;
                    redirection of stdin and stdout now *appears* to
                    work correctly in all
                    cases.&nbsp; The few PHYLIP programs distributed
                    with BioEdit now have
                    stdin redirected so they may be run fully automated.</font></li>
              </ol>
              <li>
                <font color="#000000">Forgot to report in an earlier
                  version: a bug in
                  converting selected regions to lowercase was fixed
                  (left out 'A')</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.3:</font></b>
            <ul>
              <li>
                <font color="#000000">The on-line help system was
                  updated (the printable
                  documentation still needs updating -- also the WWW
                  documaentation needs
                  to be updated (don't know when that'll get done)).</font></li>
              <li>
                <font color="#000000">Fixed a bug in resizing split
                  windows (introduced
                  with version 4.5.3 -- would cause an exception when
                  the window was increased
                  vertically due to an oversight in creating a new
                  graphic sequence display
                  object).</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.2:</font></b>
            <ul>
              <li>
                <font color="#000000">Sequence handling and memory
                  allocation for sequences
                  was re-written.&nbsp; Alignments load much faster and
                  editing of large
                  alignments is more reasonable.&nbsp; When using the
                  new BioEdit Project
                  file format, the prokaryotic 16S rRNA alignment (6205
                  sequences, 29 Mb
                  file) opens in less than 10 sec. on a Pentium 233 MHz
                  with 80 Mb RAM.&nbsp;</font></li>
              <li>
                <font color="#000000">Expanded sequence limit to 20,000.</font></li>
              <li>
                <font color="#000000">Created a new binary file format
                  (the BioEdit Project
                  file) for fast Open and Save of large alignments.</font></li>
              <li>
                <font color="#000000">A bug in formatting when exporting
                  an aligment as
                  raw text was fixed (the spacing would be randomly
                  messed up -- this bug
                  was introduced with version 4.3.8 and was only
                  recently discovered because
                  this feature apparently does not get used much).&nbsp;</font></li>
              <li>
                <font color="#000000">Two bugs in the vertical split
                  window view were fixed:&nbsp;</font></li>
              <ol>
                <li>
                  <font color="#000000">When the original window's
                    vertical scrollbar was
                    not at the top, the right side of the split window
                    would start out of register
                    with the left side.</font></li>
                <li>
                  <font color="#000000">The left side of the window
                    would not scroll all
                    the way to the end.</font></li>
              </ol>
              <li>
                <font color="#000000">A couple of changes to the
                  sequence edit box:</font></li>
              <ol>
                <li>
                  <font color="#000000">The first digit of ruler
                    positions &gt;1,000,000 is
                    no longer deleted from the ruler bar.</font></li>
                <li>
                  <font color="#000000">The "Delete" key now works in
                    the edit box as well
                    as "Backspace".</font></li>
              </ol>
              <li>
                <font color="#000000">A typo on the vertical axis of
                  Eisenberg mean hydrophobic
                  moment plots was fixed.</font></li>
              <li>
                <font color="#000000">Changes to matrix plotter:</font></li>
              <ol>
                <li>
                  <font color="#000000">Clicking on the plot when there
                    is no 1-D line plot
                    open no longer crashes the program</font></li>
                <li>
                  <font color="#000000">Highlighting of the selected
                    matrix position on the
                    plot has been fixed (the selected position is
                    inverted and outlined).</font></li>
                <li>
                  <font color="#000000">The data summary reported for a
                    highlighted matrix
                    position reflects the position numbers in the matrix
                    file rather than the
                    absolute row or column number (x and y position
                    numbers may be discontinuous,
                    as when alignment numbering is used to generate the
                    M(xy) data or when
                    a numbering mask different from the sequence mask is
                    used).</font></li>
                <li>
                  <font color="#000000">The row or column numbers shown
                    on the 1-D matrix
                    row/column plots reflect the position numbers in the
                    matrix file.</font></li>
                <br>
                <font color="#000000">A bug in reporting of the current
                  row or column
                  that showed up when an asymmetric section of a matrix
                  file was plotted
                  has been fixed (if a matrix contained more rows than
                  columns, row numbers
                  beyond the total number of columns would be
                  mis-reported).</font>
              </ol>
              <li>
                <font color="#000000">"Flags" are now available to
                  graphically mark any
                  position in an alignment for keeping track of
                  positions while when going
                  back and forth a lot.</font></li>
              <li>
                <font color="#000000">The sequences have been crunched
                  together slightly
                  for a tighter view with more sequence data on the
                  screen at once.</font></li>
              <li>
                <font color="#000000">User may enable or disable
                  position reporting at
                  the mouse arrow.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.5.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Added option to translate selected
                  regions of nucleic
                  acid sequencesfor formatted translation with codon
                  summary, with the option
                  to display the entire nucleic acid sequence or only
                  the translated region</font></li>
              <li>
                <font color="#000000">Added option for single-letter or
                  three-letter codes
                  for amino acids in formatted translations</font></li>
              <li>
                <font color="#000000">Fixed a couple of bugs related to
                  non-U.S. versions
                  of Windows: errors resulting from the difference
                  between U.S. and European
                  decimal separators (period vs comma) have been fixed
                  in the text print,
                  RNA anaylises outputs, the matrix plotter, and
                  formatted translations.&nbsp;</font></li>
              <li>
                <font color="#000000">Updated matrix plotter to provide
                  a summary of matrix
                  dimensions before loading up a matrix, and added the
                  capability to plot
                  user-defined regions of matrices.&nbsp; The plotter
                  can now handle a nearly
                  unlimited matrix size (but can still only plot ca.
                  1500 x 1500 (maybe up
                  to about 2000 x 2000) at one time.&nbsp; The largest
                  file tested was 179
                  Mb (5183 x 5183).</font></li>
              <li>
                <font color="#000000">Expanded sequence limit to 10000
                  (sequence size is
                  limited only by system resources and speed).&nbsp;</font></li>
            </ul>
            <b><font color="#000000">Bug fixes/changes since v4.5.0:</font></b>
            <ul>
              <li>
                <font color="#000000">Vertical split window view was
                  added</font></li>
              <li>
                <font color="#000000">About 16 billion bugs were fixed
                  in the split window
                  view synchronization</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.4.2:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Option added to split the
                  alignment window to simulataneously
                  work on two different parts of the same file while
                  keeping the two views
                  synchronized (menu option=View-&gt;Split Window)</font></li>
              <li>
                <font color="#000000">ORF searching (menu
                  option=Sequence-&gt;Nucleic Acid-&gt;Find
                  Next ORF) -- change preferences for ORF searching
                  under Options-&gt;Preferences-&gt;ORFs)
                  -- user-defined starts and stops and minimum length</font></li>
              <li>
                <font color="#000000">New edit box window:&nbsp; Edit
                  single sequence in
                  color with an improved sequence view including ruler
                  bar and sequence spaced
                  out in blocks of 10.</font></li>
              <li>
                <font color="#000000">Improved alignment shading utility
                  -- shade identities
                  and similarities using the current color table or
                  user-defined identity
                  and similarity font and box colors.&nbsp; Display
                  sequences in color or
                  black and white (shaded views may also be done with
                  sequences in color
                  now).&nbsp; Copy shaded alignment to clipboard as an
                  enhanced metafile
                  for direct transfer as high-quality graphic into
                  programs such as PowerPoint
                  (text-quality, place directly onto a slide background
                  or template).</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.4.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">blastx was added to the local
                  BLAST interface</font></li>
              <li>
                <font color="#000000">Image Export (Bitmap or Windows
                  Metafile) added to
                  plasmid utility with variable output size options</font></li>
              <li>
                <font color="#000000">Clipboard copy as a Windows
                  Metafile added to plasmid
                  utility and to shaded graphic view utility</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.4.0:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed "Cut" and "Copy" options for
                  working with linked
                  nucleotide/protein views.</font></li>
              <li>
                <font color="#000000">Added user control over menu
                  shortcuts in alignment
                  documents (to customize shortcuts, go to "Customize
                  Menu Shortcuts" from
                  the "View" menu)</font></li>
              <li>
                <font color="#000000">Added "Auto-Update View Options"
                  item under the "View"
                  menu -- when checked, the "Save Options as Default"
                  item is not visible,
                  and view options are automatically saved whenever a
                  change is made</font></li>
              <li>
                <font color="#000000">Fixed redraw of ends of sequences
                  when in a non-typeset
                  font.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.10:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed bug in translating and
                  translation toggling
                  (no longer bombs out after several translations)</font></li>
              <li>
                <font color="#000000">Multi-file handling for opening
                  and importing sequence
                  files</font></li>
              <li>
                <font color="#000000">Non-typeset font handling was
                  improved for the editing
                  window and in graphic shaded view</font></li>
              <li>
                <font color="#000000">Read filters for NBRF/PIR and
                  Phylip 4 formats were
                  improved</font></li>
              <li>
                <font color="#000000">Monochrome sequence view fixed
                  (can't remember when
                  or how I broke it -- but it got broken).</font></li>
              <li>
                <font color="#000000">Three options were added for
                  handling gaps for translating
                  when toggling between nucleotide and protein views:</font></li>
              <ol>
                <li>
                  <font color="#000000">Gaps automatically adjusted to
                    occur in groups of
                    three that do not disrupt codons</font></li>
                <li>
                  <font color="#000000">Gaps that disrupt codons or do
                    not occur in groups
                    of three may be ignored in trnaslating, but no
                    automatic adjustment of
                    gaps</font></li>
                <li>
                  <font color="#000000">No automatic adjustments to gaps
                    made, and gaps not
                    ignored when translating<b>&nbsp;</b></font></li>
              </ol>
              <li>
                <font color="#000000">File extensions for GenBank,
                  Fasta, PHYLIP, NBRF/PIR
                  and BioEdit Plasmid files are registered for
                  point-and-click document opening
                  from Windows&nbsp;</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.9:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Two bugs that were introduced into
                  version 4.3.9
                  were fixed:</font></li>
              <ol>
                <li>
                  <font color="#000000">Access Violation that would
                    occur after previewing
                    "Print Alignment as text"</font></li>
                <li>
                  <font color="#000000">Access Violation that would
                    occur upon running mutual
                    information analysis</font></li>
              </ol>
            </ul>
            <font color="#000000">&nbsp;&nbsp;&nbsp;&nbsp; * These two
              bugs only occurred
              in version 4.3.9 *</font>
            <br>
            &nbsp;
            <p><b><font color="#000000">Bug fixes/changes since v4.3.8:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">The program was restructured
                  internally to use less
                  memory and much fewer system resources.&nbsp; As a
                  result, the program
                  loads much faster and is not as demanding of the
                  operating system.&nbsp;
                  If you have experienced any trouble with starting
                  BioEdit, this may solve
                  the problem.</font></li>
              <li>
                <font color="#000000">Two bugs were fixed in the display
                  of selected residues:</font></li>
              <ol>
                <li>
                  <font color="#000000">The entire selection is now
                    highlighted correctly
                    in very long stretches of selected residues (more
                    than ca. 25000 residues).</font></li>
                <li>
                  <font color="#000000">When selecting entire columns by
                    selecting on the
                    sequence position ruler, the selection highlighting
                    would sometimes get
                    broken, with some selections not appearing
                    highlighted. This has been fixed.&nbsp;</font></li>
              </ol>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.7:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Added click-select capability fo
                  viewing data points
                  in the matrix plotter</font></li>
              <li>
                <font color="#000000">When creating a line graph from
                  matrix data, the
                  row selected on the matrix plotter is graphed</font></li>
              <li>
                <font color="#000000">When toggling between row and
                  column views in matrix
                  row/column graphs, the selected data point on the
                  matrix plotter is used</font></li>
              <li>
                <font color="#000000">When rows or columns are scrolled
                  through on a line
                  graph, the selected data point on the matrix plotter
                  (if there is one)
                  is&nbsp; updated to match</font></li>
              <li>
                <font color="#000000">Rows or column may be selected for
                  an open matrix
                  line graph by selecting a point on the matrix plotter</font></li>
              <li>
                <font color="#000000">Fixed a bug in memory allocation
                  for the matrix plotter:&nbsp;
                  The last point of the matrix no longer disappears</font></li>
              <li>
                <font color="#000000">Matrix table reading is now more
                  flexible (It can
                  handle inappropriate spaces, tabs or returns in the
                  file without giving&nbsp;
                  an error)</font></li>
              <li>
                <font color="#000000">Matrix plotter now reads .csv
                  (comma-delimited) matrices
                  as well as tab-delimited</font></li>
              <li>
                <font color="#000000">A bug was fixed for updating
                  feature titles in the
                  plasmid drawing utility:&nbsp; The updated title is
                  selected and sized
                  correctly on the screen now</font></li>
              <li>
                <font color="#000000">Linear vector maps should now
                  shade better on a grayscale
                  laser printer (only circular maps were fixed
                  previously)</font></li>
              <li>
                <font color="#000000">Moving the ends of horizontal or
                  vertical arrows
                  in the plasmid drawing utility now works correctly
                  (doesn't get stuck horizontal
                  or vertical anymore)</font></li>
              <li>
                <font color="#000000">A bug in the new
                  nucleotide/protein translation view
                  toggling was fixed:&nbsp; Previously, if gaps were
                  accidentally introduced
                  into the last codon of a nucleotide sequence and
                  several gaps also occurred
                  at the end of the sequence, an error would occur when
                  BioEdit attempted
                  to readjust gaps to mend codons.&nbsp; This should no
                  longer be a problem</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.6:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">The on-line help system was
                  updated with a full searchable
                  index and full text search capability</font></li>
              <li>
                <font color="#000000">Nucleotide sequences which encode
                  proteins may be
                  switched between their protein and nucleic acid
                  sequences, and alignments
                  may be edited in either view</font></li>
              <li>
                <font color="#000000">Simple translations of nucleic
                  acids and reverse
                  translations of proteins now take gaps into account
                  rather than treating
                  them as unknown residues</font></li>
              <li>
                <font color="#000000">ClustalW may now be run without
                  modifying sequence
                  titles to be non-redundant and 10 or few characters
                  long</font></li>
              <li>
                <font color="#000000">When ClustalW is run from the
                  accessory menu, the
                  new alignment is automatically updated with GenBank
                  information (features,
                  references, accession, etc.) and full titles</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.5:</font></b>
              <br>
              <font color="#000000">(These changes were both made to the
                plasmid
                drawing utility)</font>
            </p>
            <ul>
              <li>
                <font color="#000000">Grouped drawing objects now scale
                  as a unit</font></li>
              <li>
                <font color="#000000">Changed feature drawing (arrows
                  and boxes) to be
                  more compatible with most printers</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.4:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Added linear vector capability</font></li>
              <li>
                <font color="#000000">Made titles selectable and
                  editable</font></li>
              <li>
                <font color="#000000">Made vector structure selectable
                  and mouse-movable</font></li>
              <li>
                <font color="#000000">All objects and labels move along
                  with vector when
                  moving by hand</font></li>
              <li>
                <font color="#000000">Restriction sites and feature
                  labels locations adjust
                  themselves when vector is resized&nbsp;</font></li>
              <li>
                <font color="#000000">Bug fixed in plasmid
                  utility:&nbsp; selecting multiple
                  restriction site labels then hitting "delete" was
                  causing trouble with
                  the enzyme-label links</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.3:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">A bug in the positioning of
                  restriction cut sites
                  at the origin junction of circular DNA sequences was
                  fixed.&nbsp; This
                  error occured when running a map on circular sequences
                  that had a restriction
                  site&nbsp; that overlapped the origin and cut after
                  the overlap point.&nbsp;
                  This would cause an error also when generating a
                  plasmid map from such
                  a sequence.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.2:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Only one change:&nbsp; The
                  sequence editing box now
                  behaves better, and there is no set limit on the size
                  of the sequence that
                  can be opened and edited in an edit box (well,
                  actually the theoretical
                  upper limit is10 million bases, but that would be too
                  slow to be worth
                  it, anyway).</font></li>
              <li>
                <font color="#000000">The printable documentation has
                  been updated (It's
                  only available for MS Word'97 or later).</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.3.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed an error in Kyte and
                  Doolittle hydrophobicity
                  data (valine was entered as -4.2 instead of 4.2)</font></li>
              <li>
                <font color="#000000">Fixed enzyme label-position
                  hot-linking in plasmid
                  drawing utility</font></li>
              <li>
                <font color="#000000">Enzyme labels now directly linked
                  by lines to their
                  respective positions in the plasmid drawing utility.</font></li>
              <li>
                <font color="#000000">Scaling to the printer fixed for
                  plasmid drawing</font></li>
              <li>
                <font color="#000000">Added arrows to drawing tools in
                  plasmid drawing
                  utility.</font></li>
              <li>
                <font color="#000000">Copy and Paste now works between
                  plasmid documents
                  and toolbar is updated correctly when switching
                  between multiple plasmid
                  documents (was a problem before).</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.1.2:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Modified hydrophobicity plotting
                  to conform to Kyte
                  and Doolittle mean hydrophobicity profile.</font></li>
              <li>
                <font color="#000000">Added file save and open
                  capability for plasmid drawing
                  files</font></li>
              <li>
                <font color="#000000">Added direct print for plasmid
                  drawings</font></li>
              <li>
                <font color="#000000">Added object property editing for
                  plasmid drawing</font></li>
              <li>
                <font color="#000000">Added copy and paste functions for
                  plasmid drawing
                  and modified copy to allow copying objects as bitmaps
                  for pasting into
                  other apps</font></li>
              <li>
                <font color="#000000">Added object ordering support in
                  plasmid drawing
                  utility</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v4.1.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Several bug fixes in new plasmid
                  mapping utility:</font></li>
              <ul>
                <li>
                  <font color="#000000">no longer crashes when creating
                    a vector from a sequence
                    containing lowercase letters</font></li>
                <li>
                  <font color="#000000">Extra restriction enzymes no
                    longer show up in restriction
                    enzyme dialog box</font></li>
                <li>
                  <font color="#000000">Feature and enzyme labels now
                    are created as objects
                    that may be moved and scaled</font></li>
                <li>
                  <font color="#000000">features may now cross the orign
                    (allowing proper
                    control over direction)</font></li>
                <li>
                  <font color="#000000">Scaling drawing objects works
                    more intuitively now</font></li>
                <li>
                  <font color="#000000">Hand-drawn lines behave as lines
                    rather than recangular
                    diagonals (only the ends are selected, rather than a
                    rectangular bounding
                    region, and ends may be moved around leaving the
                    opposite end anchored)</font></li>
              </ul>
              <p><br>
                <font color="#000000">The plasmid drawing utility has a
                  long way
                  to go.&nbsp; Things that need to be added or improved
                  include:</font>
              </p>
              <ul>
                <li>
                  <font color="#000000">More drawing tools</font></li>
                <li>
                  <font color="#000000">Support for rotating objects</font></li>
                <li>
                  <font color="#000000">Editing object properties after
                    drawing them on the
                    screen</font></li>
                <li>
                  <font color="#000000">Direct print to the printer
                    (copying to the clipboard
                    produces screen-resolution, very pixellated, images)</font></li>
                <li>
                  <font color="#000000">Plasmid file format the allows
                    the user to save a
                    plasmid document as an editable, vectored graphic</font></li>
                <li>
                  <font color="#000000">Some sort of documentation
                    (documentation for the
                    whole program badly needs to be updated)</font></li>
              </ul>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v3.1.2:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Added plasmid drawing interface
                  with annotation tools
                  and polylinker view -- this is in an early stage and
                  is not yet finished.&nbsp;
                  It may also have a couple of bugs.</font></li>
              <li>
                <font color="#000000">Fixed bug in multipage graphical
                  alignment views
                  (no longer crashes on the last page -- also made
                  graphic view more efficient)</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v3.1.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed bugs in opening of Fasta
                  files from Mac or
                  UNIX format (was losing the first line of
                  sequence)&nbsp;</font></li>
              <li>
                <font color="#000000">Fixed minor bug in opening of
                  NBRF/PIR sequences
                  (spaces in the title would truncate the title and tack
                  extra characters
                  at the beginning of the sequence&nbsp;</font></li>
              <li>
                <font color="#000000">Fixed color matching between
                  series lines and legend
                  of hydrophobicity plots&nbsp;</font></li>
              <li>
                <font color="#000000">Added scrolling/keyboard control
                  of row/column number
                  in line plots of mutual information matrix data</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v3.0.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Fixed font handling in edit window
                  -- any fonts will
                  now scale correctly with ruler and selections.</font></li>
              <li>
                <font color="#000000">Added
                  hydrophobicity/hydrophobicity profiles for
                  proteins(Kyte-Doolittle, Eisenberg, Cornette, Boyko,
                  Parker, and Hopps-Woods
                  scales).</font></li>
              <li>
                <font color="#000000">Added amino acid and nucleotide
                  composition summaries
                  and plots.</font></li>
              <li>
                <font color="#000000">Reorganized the "Sequence" menu
                  (now has separate
                  "Nucleic Acids" and "Protein" menus)</font></li>
              <li>
                <font color="#000000">Fixed bug in Mask toggling (would
                  cause an exception
                  if selection went beyond the end of the mask)</font></li>
              <li>
                <font color="#000000">Added restriction enzyme
                  mapping/browsing by manufacturer</font></li>
              <li>
                <font color="#000000">Updated restriction enzyme table
                  with the REBASE
                  gcgenz.811 table</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v2.1.1:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Added zoom capability and forced
                  data point size
                  to data matrix plotter</font></li>
              <li>
                <font color="#000000">Added line/area graphing for rows
                  of mutual information
                  matrix data with point-and-click data examination</font></li>
              <li>
                <font color="#000000">Experimental three-way information
                  (M(xyz), M(x,yz),
                  M(xy,z) and M(xz,y)) -- doesn't seem to be very
                  worthwhile, however.</font></li>
              <li>
                <font color="#000000">Added print and clipboard copy
                  capabilities to two-dimensional
                  graphs (so far only Entropy plots and data matrix row
                  plots are available).</font></li>
              <li>
                <font color="#000000">BioEdit is now distributed with
                  the <a
                    href="http://taxonomy.zoology.gla.ac.uk/rod/rod.html">TreeView</a>
                  installation package which may be installed
                  independently of BioEdit.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v2.0.4:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">fixed bug in LOCUS field of
                  GenBank files (was duplicationg
                  part of the LOCUS in some cases</font></li>
              <li>
                <font color="#000000">Better handling of
                  Macintosh-formatted GenBank files</font></li>
              <li>
                <font color="#000000">Automatic scrolling in Grab and
                  Drag mode.</font></li>
              <li>
                <font color="#000000">Arrow key control over window
                  scrolling</font></li>
              <li>
                <font color="#000000">Added horizontal scroll speed
                  controller</font></li>
              <li>
                <font color="#000000">Fixed information-responsive
                  residue shading (capital/lowercase
                  residues no longer seen as different, and multiple
                  gaps now reduce the
                  alignment strength rather than increase it)&nbsp;</font></li>
              <li>
                <font color="#000000">Uppercase/lowercase, lock/unlock
                  and degap control
                  of sequence selections as well as complete sequences.</font></li>
            </ul>
            <p><br>
              <b><font color="#000000">Bug fixes/changes since v2.0.3:</font></b>
            </p>
            <ul>
              <li>
                <font color="#000000">Automatic scrolling in document
                  while selecting residues:
                  Scrolling past the end no longer causes the beginning
                  of the selection
                  to be moved back, and scrolling paste the beginning no
                  longer causes the
                  window to scroll forward.</font></li>
              <li>
                <font color="#000000">Copy/paste: Using the normal copy
                  function will now
                  copy entire sequence structures as long as no residues
                  are selected on
                  the screen. Paste works just like Paste Sequence(s) as
                  long as the document
                  is not in Edit Residues mode. Copy Sequence(s) will
                  both copy sequence
                  structures to the BioEdit clipboard and also
                  Fasta-formatted sequences
                  to the Windows clipboard.&nbsp;</font></li>
              <li>
                <font color="#000000">The New Alignment option of the
                  File menu now works
                  from within an alignment document.</font></li>
              <li>
                <font color="#000000">Consensus sequence generation no
                  longer drops the
                  last residue</font></li>
              <br>
              <font color="#000000">Degapping and unlocking gaps does
                not cause an
                error when selection(s) go paste the end of a sequence.</font>
            </ul>
          </td>
        </tr>
        <tr>
          <td><br>
          </td>
          <td><font color="#000000"><a href="mailto:tomandhall@gmail.com">tomandhall@gmail.com</a></font>
            <p><font color="#000000"><a href="https://thalljiscience.github.io">Back to
                  BioEdit Main Page</a></font>
              <br>
          </td>
        </tr>
      </tbody>
    </table>
  </body>
</html>
